src/syft/grid/duet/__init__.py

```diff
@@ -134,7 +134,8 @@ def run(self) -> None:
                     sys.stdout.write("\r" + out)
                 iterator += 1
 
-    counterThread().start()
+    if hasattr(sys.stdout, "parent_header"):
```

Comment:
Q: Does this means that the duet will run only inside a jupyter notebook?

Reply:
It should run fine, but there will be no live status update with # of messages/etc as in the notebook